hashing/keccak/keccak.go

```diff
@@ -5,38 +5,42 @@ import (
 	"golang.org/x/crypto/sha3"
 )
 
-var _ hashing.Hasher = (*Keccak)(nil)
+var _ hashing.Hasher = (*keccak)(nil)
 
-var keccakEmptyHash []byte
+// keccak is a sha3-keccak implementation of the hasher interface.
+type keccak struct {
+	emptyHash []byte
+}
 
-// Keccak is a sha3-Keccak implementation of the hasher interface.
-type Keccak struct {
+// NewKeccak initializes the empty hash and returns a new instance of the keccak hasher
+func NewKeccak() *keccak {
+	return &keccak{
+		emptyHash: computeEmptyHash(),
+	}
 }
 
-// Compute takes a string, and returns the sha3-Keccak hash of that string
-func (k Keccak) Compute(s string) []byte {
-	if len(s) == 0 && len(keccakEmptyHash) != 0 {
-		return k.EmptyHash()
+// Compute takes a string, and returns the sha3-keccak hash of that string
+func (k *keccak) Compute(s string) []byte {
+	if len(s) == 0 {
+		return k.emptyHash
```

Comment:
please return a copy of emptyHash instead

Reply:
done